compiler/src/dotty/tools/dotc/core/tasty/TreePickler.scala

```diff
@@ -409,22 +409,24 @@ class TreePickler(pickler: TastyPickler) {
             case _ =>
               val sig = tree.tpe.signature
               var ename = tree.symbol.targetName
-              val isAmbiguous =
-                sig != Signature.NotAMethod
-                && qual.tpe.nonPrivateMember(name).match
-                  case d: MultiDenotation => d.atSignature(sig, ename).isInstanceOf[MultiDenotation]
-                  case _ => false
-              if isAmbiguous then
+              val selectFromQualifier =
+                name.isTypeName
+                || qual.isInstanceOf[TreePickler.Hole] // holes have no symbol
+                || sig == Signature.NotAMethod // no overload resolution necessary
+                || !tree.denot.symbol.exists // polymorphic function type
+                || !tree.denot.isOverloaded // refined methods are never overloads, they have no defining class
```

Comment:
Is this check needed? I thought `isOverloaded` always returned false after Typer because overloading resolution was finished.

Reply:
so everything should be guaranteed to be `SingleDenotation`?

Reply:
Yes, this is even checked since only SingleDenotation have symbols: https://github.com/lampepfl/dotty/blob/8ae56118e6accf4ce870b8717b8cf966240c602f/compiler/src/dotty/tools/dotc/transform/TreeChecker.scala#L383

Reply:
nice, I'll remove that